### ISSUE TITLE
Finish initial migration

### ISSUE THREAD

**Issue by jperryhouts** [Extended-Thunder/send-later#6]
A number of minor-ish things need to happen before the migration is usable. As of now, all of the basic functions are implemented, but not all wired up to one another. UI is not yet connected to the backend in many places, code doesn't always respect user preferences, and recurrence is not fully functional.

High priority:  
- [x] Implement "send at" logic for popup dialog
- [x] Un-schedule messages that are presently being edited. (Commit 972b9f2)
- [x] Encode/decode recurrence specifications (Commit 7136ef9)
- [x] Respect limits ("between", "only on", late msg blocker) (Commits 5a34c9e and  972b9f2)
- [x] User-defined recurrence functions (Commits b61f178782010d71bf4815ba4a510eadbfb07520, e18629cdd4c069f3507bde8a2ebd902156819a6a, and 8a0da8dfcb6061930feb8db0e152c2d08d8f14e5)
- [x] Handle case where thunderbird is currently offline (Commit 5a34c9e)

Lower priority:  
- [x] Cancel recurrence on reply (Commit 4005249eeddb165931d0a9897ae4aaaffc7887cb)
- [x] Mark draft messages as read (Commit 90ebe84)
- [x] Show send-later headers in messageDisplay tabs (Commit 6ab346c3eb86401053abb0748a816ad7fc0a2852)
- [x] Implement optional bind to Alt-Shift-Enter shortcut (Commit 70903eb)
- [x] Option to save popup dialog selections as defaults (Commit 0c799b5)
- [x] More complex editable shortcut buttons (Commit 9d8cf3723f5032d05573626d6508173d69b07d0a)
- [ ] Ask before quitting if there are pending messages to be sent

Lowest priority:
- [x] Check spelling and subject line before scheduling message (Commit 2c0d715c358733ac3c061ef46996c6386a32d923)
- [ ] Show "Send Later" column in draft folder
- [ ] Display process info in status bar
- [ ] Optionally log to file in addition to console
- [ ] Remove unused option variables
- [ ] Clean up unused code
- [ ] Comments and documentation
- [ ] User guide via Github wiki
